core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java

```diff
@@ -600,8 +601,13 @@ private void saveGatewayPillar(GatewayConfig gatewayConfig, Cluster cluster,
 
         ExposedService rangerService = exposedServiceCollector.getRangerService();
         List<String> rangerLocations = serviceLocations.get(rangerService.getServiceName());
+
         if (!CollectionUtils.isEmpty(rangerLocations)) {
-            serviceLocations.put(rangerService.getServiceName(), getSingleRangerFqdn(gatewayConfig.getHostname(), rangerLocations));
+            if (SdxClusterShape.LIGHT_DUTY.equals(cluster.getStack().getSdxClusterShape())) {
+                serviceLocations.put(rangerService.getServiceName(), getSingleRangerFqdn(gatewayConfig.getHostname(), rangerLocations));
+            } else if (SdxClusterShape.MEDIUM_DUTY_HA.equals(cluster.getStack().getSdxClusterShape())) {
+                serviceLocations.put(rangerService.getServiceName(), rangerLocations.stream().filter(s -> s.contains("gateway")).collect(Collectors.toList()));
```

Comment:
this doesn't seems to be a good long term solution. I would suggest to fetch the gateways' hostnames/fqdns and filter in them instead of the magic `gateway` string

Reply:
The gateways are called gateway in Medium duty.  But I can change it to that instead.

Reply:
I think you can filter on `stack.getPrimaryGatewayInstance().getInstanceGroup().getGroupName()`. For medium duty the group name should be "gateway", and for light duty it would be "master".